sway-lib-std/src/string.sw

```diff
@@ -5,38 +5,109 @@ use ::convert::From;
 use ::option::Option;
 use ::assert::assert;
 
-/// A UTF-8 encoded growable string
-///
-/// WARNING: As this type is meant to be forward compatible with UTF-8, do *not*
-/// add any mutation functionality or unicode input of any kind until `char` is
-/// implemented, codepoints are *not* guaranteed to fall on byte boundaries
+// WARNING: As this type is meant to be forward compatible with UTF-8, do *not*
```

Comment:
I'd add this as additional information to the struct rather than just a comment in the file

Reply:
But thats a warning to not add additional features, aka a warning to the library devs. People using the library dont need to see that

Reply:
I can still import the `String` type and create my own custom implementations. This will get removed down the road as well